libs/utils/utils.js

```diff
@@ -306,6 +348,38 @@ export const getFederatedUrl = (url = '') => {
   return url;
 };
 
+export function hasLanguageLinks(area) {
+  for (const link of area.querySelectorAll('a[href]')) {
+    try {
+      const { hostname } = new URL(link.href);
+      if (LANGUAGE_BASED_DOMAINS.some((domain) => hostname.includes(domain))) {
+        return true;
+      }
+    } catch {
+      // do nothing
+    }
+  }
+  return false;
+}
+
+export async function loadLanguageConfig() {
+  const parseList = (str) => str.split(/[\n,]+/).map((t) => t.trim()).filter(Boolean);
+  try {
+    const response = await fetch(`${getFederatedContentRoot()}/federal/assets/data/languages-config.json`);
```

Comment:
Verify my assumption: since this is in federal, every single consumer will use the same config?

Reply:
yes, that's my idea at the moment. I think until adobe.com is fully langauge based we need a config to let sites know which ones are language based already so that the hybrid of locale and lang links works properly. At some point this config will go away though 💪